### PR TITLE
Emit the created content as well

### PR DESCRIPTION
**Pull request by mohsen1**
Very similar to #7 but emits created contnet as well. I tried it on my setup and it works.

### CODE REVIEW

**Comment by mohsen1** [comment]
//cc @Idorobots

**Comment by olegstepura** [comment]
Hi! Thanks! Waiting for confirmation from @Idorobots.

File: index.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ module.exports = function(source, map) {
   // creator.create(..., source) tells the module to operate on the
   // source variable. Check API for more details.
   creator.create(this.resourcePath, source).then(content => {
+    // Emit the created content as well
+    this.emitFile(this.resourcePath + '.d.ts', content.contents || ['']);
```
Comment (Idorobots) [inline index.js]: I think this should use `content.outputFilePath` in order to support `outDir` query parameter.

**Comment by Idorobots** [comment]
I'm still getting an error on the first, clean build in my setup. As mentioned in #7, in my case it seems to not matter as the build continues but I'm not sure that it'll be the case for others.

**Comment by olegstepura** [comment]
@Idorobots does it work if you create a CSS file in a project after you started webpack watch?

**Comment by Idorobots** [comment]
@olegstepura, initially it fails to build the bundle, but it compiles properly after I edit a .ts file that imports this .css file again. Webpack doesn't crash, so I suppose it's usable, but I wouldn't say it _works_ either.

**Comment by mohsen1** [comment]
The way I went around it was to create a global dummy `*.css` module 

``` ts
declare module '*.css' {
    const result: any;
    export = result;
}
```

In my editor, I see TypeScript complaining if I use a class that doesn't exist in the CSS file so it means it uses the local `.css.d.ts` instead of that global module.

File: index.js (hunk 1 above)
Comment (mohsen1) [reply index.js]: `content.outputFilePath` has the `.d.ts` at the end, right?

**Comment by mohsen1** [comment]
@Idorobots updated.

Although this works very well (if you add the `*.css` module) it is emitting those files after the TypeScript compilation is completed so in the first run it will not catch issues for you.

**Comment by mohsen1** [comment]
Can we use a pitching loader to do this process instead?

**Comment by olegstepura** [comment]
@mohsen1 Is there anything you are going to add to this PR?
If not please let me know if I can merge this and release a newser version of loader.

**Comment by mohsen1** [comment]
Well, this + that module declaration works for us but if you misspell a class name it will not fail due to forgiving `*.css` module declaration. 

I'm experimenting with [pitching loaders](https://webpack.github.io/docs/loaders.html#pitching-loader). It's exactly what we're looking for. We want the metadata and don't care about input and output. We only want to produce some side effect early on. 

The only problem I have is that pitching loaders are sync and DSTCreator is async. So when I do something like this, sometimes some modules fail because the async process took more than one CPU tick...

Do you know if DST Creator can be sync?

``` js
var DtsCreator = require('typed-css-modules');
var loaderUtils = require('loader-utils');
var objectAssign = require('object-assign');

module.exports.pitch = function(fullpath) {
  this.cacheable && this.cacheable();

  var parts = fullpath.split('!');
  var name = parts[parts.length - 1];

  var creator = new DtsCreator();
  creator.create(name).then((content) => {
        content.writeFile();
  });

  return;
};
```

**Comment by mohsen1** [comment]
@Quramy how hard is it to make sync methods for DTS Creator?

**Comment by mohsen1** [comment]
oh I guess I can use `this.async` in pitching loaders too. Not sure what the callback expects there. most of things I tried just breaks the webpack process!

**Comment by olegstepura** [comment]
> Well, this + that module declaration works for us but if you misspell a class name it will not fail due to forgiving *.css module declaration.

@mohsen1 Could you be so kind to update README accordingly? So that other users would be informed.

Also I'm thinking of adding (that) module declaration to loader sources, so that it would be in a release to just include it in a project codebase.

**Comment by mohsen1** [comment]
@olegstepura Will do if that's  the final solution but if I can make the pitching loader work, it would be unnecessary to all of those hacks and it will work as anyone expect.

**Comment by mohsen1** [comment]
ok still struggling with making the pitch loader to 
- write a new file
- add it to dependency graph so TypeScript loader sees it 

I'm pretty sure `this.emitFile` is not what we want. It will produce a result, we only want to have a file to be added to the collection of files that webpack is seeing.

**Comment by olegstepura** [comment]
@mohsen1 Ok, let's wait then ;)

**Comment by olegstepura** [comment]
@mohsen1 Can this already be merged? Or do you plan to add more commits?

**Comment by mohsen1** [comment]
No this will not fix the original issue

**Comment by olegstepura** [comment]
Is it harmless to merge this?

**Comment by mohsen1** [comment]
Not necessarily. It will work better for the second run.

**Comment by olegstepura** [comment]
Sorry, I'm confused... If it's better to merge this and it will not hurt anyone, I will merge this. 
I stopped using typescrypt myself, so I'm kind of not interested in this anymore.

**Comment by mohsen1** [comment]
@olegstepura I just tried this with our CI system. Not emitting the TSDs broke our build. 

Could you please merge this change and release it?

Thanks

**Comment by olegstepura** [comment]
Merged and released v0.0.6.

**Pull request merged**